### DISCUSSION TITLE
New Swap interface and functionality suggestions

### DISCUSSION THREAD

**Discussion by jasonpearce** [Impa10r/peerswap-web#70] [Ideas]
Thank you for creating PeerSwap Web UI. I would like to make some suggestions that I hope you will consider.

For the New Swap screen, as a rookie, it was confusing at first if I was swapping in our swapping out. And I have to do some math. I was unclear if Swap-In meant swapping in Liquid or swapping in Sats. 

My suggestion is that you add an Objective pull-down menu above Channel named "Objective" with some options like these:

* Rebalance Lightning Channel to 50/50 using Bitcoin (BTC)
* Rebalance Lightning Channel to 50/50 using Liquid (L-BTC)
* Empty Lightning Channel to 10/90 using Bitcoin (BTC)
* Empty Lightning Channel to 10/90 using Liquid (L-BTC)
* Fill Lightning Channel to 90/10 using Bitcoin (BTC)
* Fill Lightning Channel to 90/10 using Liquid (L-BTC)

Depending on what the user selects, and their available balance of either L-BTC or BTC, you then automatically calculate and fill in the values for Direction, Asset, and Amount.

Or, look at the GUI of Botz (https://boltz.exchange/) and FixedFloat (https://ff.io/). Then take the best of their approaches and make your own. They make it more clear that you are Sending one asset so that you will Receive a different asset.

I appreciate all that you've done. Please consider this helpful feedback. Thank you.

**Comment by Impa10r** [comment]
> For the New Swap screen, as a rookie, it was confusing at first if I was swapping in our swapping out. And I have to do some math. I was unclear if Swap-In meant swapping in Liquid or swapping in Sats.

Hello, thank you for the feedback. An arrow to or from lightning shows the direction:

![image](https://github.com/Impa10r/peerswap-web/assets/101550606/046f8a87-a393-4eef-8e78-ba931fab86cd)

Also, the default direction is assumed from the channel balance. I will consider your suggestions how to make it clearer.

**Comment by Impa10r** [comment]
@jasonpearce Redesigned New Swap form to be more intuitive in v1.6.0. Thanks again for your feedback!

![image](https://github.com/Impa10r/peerswap-web/assets/101550606/28952cde-6129-4c16-a315-d0b9d31bdad9)

**Reply by jasonpearce** [reply]
I think what you have proposed is much easier to understand. Thank you for considering the feedback. Nice job.

My other idea was to auto calculate the amount. In this example you would have recommended 76,055 as the amount.

((222,546 - 70,435) / 2 ) = 76,055.5

A recommended amount of 76,055 would balance the channel 50/50 (ignoring fees). If you want to be more accurate, factor in the estimated fee too so that the recommended Amount is closer to a 50/50 balanced channel.

**Reply by Impa10r** [reply]
I have more or less this functionality already in AutoSwap. When you do a manual one it is usually better to use the whole available balance of L-BTC if possible. This will give you the minimal PPM cost. Trying to get L-BTC from a peer (swap out) 99.9% fails, so no one should do that.
